pkg/cassandra/datacenter.go

```diff
@@ -0,0 +1,141 @@
+	AdditionalSeeds   []string
+}
+
+func NewDatacenter(klusterKey types.NamespacedName, template *DatacenterConfig) (*cassdcapi.CassandraDatacenter, error) {
+	namespace := template.Meta.Namespace
+	if len(namespace) == 0 {
+		namespace = klusterKey.Namespace
+	}
+
+	rawConfig, err := CreateJsonConfig(template.CassandraConfig, template.ServerVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	dc := &cassdcapi.CassandraDatacenter{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   namespace,
+			Name:        template.Meta.Name,
+			Annotations: map[string]string{},
+			Labels: map[string]string{
+				api.NameLabel:             api.NameLabelValue,
+				api.PartOfLabel:           api.PartOfLabelValue,
+				api.ComponentLabel:        api.ComponentLabelValueCassandra,
+				api.CreatedByLabel:        api.CreatedByLabelValueK8ssandraClusterController,
+				api.K8ssandraClusterLabel: klusterKey.Name,
+			},
+		},
+		Spec: cassdcapi.CassandraDatacenterSpec{
+			ClusterName:     template.Cluster,
+			ServerImage:     template.ServerImage,
+			Size:            template.Size,
+			ServerType:      "cassandra",
+			ServerVersion:   template.ServerVersion,
+			Config:          rawConfig,
+			Racks:           template.Racks,
+			StorageConfig:   *template.StorageConfig,
+			AdditionalSeeds: template.AdditionalSeeds,
+			// TODO The Networking field should be exposed in DatacenterConfig
+			Networking: &cassdcapi.NetworkingConfig{
+				HostNetwork: true,
+			},
+		},
+	}
+
+	if template.Resources != nil {
+		dc.Spec.Resources = *template.Resources
+	}
+
+	return dc, nil
+}
+
+// Coalesce combines the cluster and dc templates with override semantics. If a property is
+// defined in both templates, the dc-level property takes precedence.
+func Coalesce(clusterTemplate *api.CassandraClusterTemplate, dcTemplate *api.CassandraDatacenterTemplate) *DatacenterConfig {
+	dcConfig := &DatacenterConfig{}
+
+	// Handler cluster-wide settings first
+	dcConfig.Cluster = clusterTemplate.Cluster
+
+	// DC-level settings
+	dcConfig.Meta = dcTemplate.Meta
+	dcConfig.Size = dcTemplate.Size
+
+	if len(dcTemplate.ServerVersion) == 0 {
+		dcConfig.ServerVersion = clusterTemplate.ServerVersion
+	} else {
+		dcConfig.ServerVersion = dcTemplate.ServerVersion
+	}
+
+	if len(dcTemplate.ServerImage) == 0 {
+		dcConfig.ServerImage = clusterTemplate.ServerImage
+	} else {
+		dcConfig.ServerImage = dcTemplate.ServerImage
+	}
+
+	if len(dcTemplate.Racks) == 0 {
+		dcConfig.Racks = clusterTemplate.Racks
+	} else {
+		dcConfig.Racks = dcTemplate.Racks
+	}
+
+	if dcTemplate.Resources == nil {
+		dcConfig.Resources = clusterTemplate.Resources
+	} else {
+		dcConfig.Resources = dcTemplate.Resources
+	}
+
+	// TODO Add validation check to ensure StorageConfig is set at the cluster or DC level
+	if dcTemplate.StorageConfig == nil {
+		dcConfig.StorageConfig = clusterTemplate.StorageConfig
+	} else {
+		dcConfig.StorageConfig = dcTemplate.StorageConfig
+	}
+
+	// TODO Do we want merge vs override?
```

Comment:
Merge.

Reply:
I will create a separate ticket to do this.

Reply:
I created #122